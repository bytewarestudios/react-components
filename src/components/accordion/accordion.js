import React from 'react';
import './accordion.css'

export class Accordion extends React.Component {
  // mock data this could come from an api.
  accordion = [
    {
      id: 'accordion-id-1',
      title: 'accordion title 1',
      content: 'I am the contents of tab 1'
    },
    {
      id: 'accordion-id-2',
      title: 'accordion title 2',
      content: 'I am the contents of tab 2'
    }
  ];

  constructor(props) {
    super();
  }

  toggleAccordion(event) {
    const accordionHeader = document.querySelectorAll('#accordion div h2')

    this.accordion.forEach((accordionTab, i) =>
      event.target.id === accordionTab.id ?
        accordionHeader[i].nextElementSibling.classList.value.indexOf('closed') > -1
        ? accordionHeader[i].nextElementSibling.setAttribute('class', '')
        : accordionHeader[i].nextElementSibling.setAttribute('class', 'closed')
        : null
    );
  }

  render() {
    let accordions  = this.accordion.map((accordion) =>
      <div key={accordion.id}>
        <h2 onClick={this.toggleAccordion.bind(this)} id={accordion.id}>{accordion.title}</h2>
        <div className="closed">{accordion.content}</div>
      </div>
    );

    return <div id="accordion">{accordions}</div>;
  }
}
