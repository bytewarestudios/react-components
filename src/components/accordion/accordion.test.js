import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { Accordion } from './accordion';

const accordion = [
  {
    id: 'accordion-id-1',
    title: 'accordion title 1',
    content: 'I am the contents of tab 1'
  },
  {
    id: 'accordion-id-2',
    title: 'accordion title 2',
    content: 'I am the contents of tab 2'
  }
];


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Accordion />, div);

})


it('renders a snapshot', () => {
  const tree = renderer.create(<Accordion />).toJSON();
  console.log(tree);
  // expect(tree).toMatchSnapshot();
});


