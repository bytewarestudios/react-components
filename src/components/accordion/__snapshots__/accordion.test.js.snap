// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  id="accordion"
>
  <div>
    <h2
      id="accordion-id-1"
      onClick={[Function]}
    >
      accordion title 1
    </h2>
    <div
      className="closed"
    >
      I am the contents of tab 1
    </div>
  </div>
  <div>
    <h2
      id="accordion-id-2"
      onClick={[Function]}
    >
      accordion title 2
    </h2>
    <div
      className="closed"
    >
      I am the contents of tab 2
    </div>
  </div>
</div>
`;
